lqsim/lqsim.1
.TH LQSIM 1 "$Date::2013-01-30 $"
.\" $Id: lqsim.1 17458 2024-11-12 11:54:17Z greg $
.SH NAME
lqsim \- simulate layered queueing networks. 
.SH SYNOPSIS
.sp
.B lqsim
[
.B \-dnpRvV
] [
.B \-A \fIrun-time[,precision[,skip]]\fR
] [
.B \-B \fIblocks[,run-time[,skip]]\fR
] [
.B \-C \fIprecision[,initial-loops[,run-time]]\fR
] [
.B \-e \fI(e|i|w)\fR
] [
.B \-h \fI(file|directory)\fR
] [
.B \-I \fI(xml|lqn)\fR
] [
.B \-m \fIfile\fR
] [
.B \-o \fI(file|directory)\fR
] [
.B \-P \fIpragma\fR
] [
.B \-S \fIseed\fR
] [
.B \-t \fItraceopts\fR
] [
.B \-T \fIrun-time\fR
] [
filename
\&.\|.\|.
]
.SH DESCRIPTION
\fBLqsim\fR is used to simulate layered queueing networks using the
PARASOL simulation system.  Models can be specified using the LQN
modelling language, or with extensible markable language (XML).  Refer
to the \fI``Layered Queueing Network Solver and Simulator User
Manual''\fR for details of the model and for a complete description of
the input file formats for the program.
.PP
\fBLqsim\fR reads its input from \fIfilename\fR, specified at the
command line if present, or from the standard input otherwise.  By
default, output for an input file \fIfilename\fR specified on the
command line will be placed in the file \fIfilename\fB.out\fR.  If the
\fB\-p\fP switch is used, parseable output will also be written into
\fIfilename\fB.p\fR. If XML input or the \fB\-x\fP switch is used, XML output will be written
to \fIfilename\fB.lqxo\fR.  This behaviour can be changed using
the \fB\-o\fIoutput\fR switch, described below.  If several files are
named, then each is treated as a separate model and output will be
placed in separate output files.  If input is from the standard input,
output will be directed to the standard output.  The file name `\-' is
used to specify standard input.
.PP
The \fB\-o\fIoutput\fR option can be used to direct output to the file
or directory named \fIoutput\fR regardless of the source of input.
Output will be XML if XML input or if the \fB\-x\fP switch is used,
parseable output if the \fB\-p\fP switch is used, and normal output
otherwise; multiple input files cannot be specified.  If \fIoutput\fP
is a directory, results will be written in the directory named
\fIoutput\fP.  Output can be directed to standard output by using
\fB\-o\fI\-\fR (i.e., the output file name is `\fI\-\fR'.)
.SH "OPTIONS"
.TP
\fB\-A\fR, \fB\-\-automatic\fR=\fIrun-time[,precision[,skip]]\fR
Use automatic blocking with a simulation block size of \fIrun-time\fP.
The \fIprecision\fP argument specifies the desired mean 95% confidence
level.  By default, \fIprecision\fP is 1.0%.  The simulator will stop
when this value is reached, or when 30 blocks have run.  \fISkip\fP
specifies the time value of the initial skip period.  Statistics
gathered during the skip period are discarded.  By default, its value
is 0.  When the run completes, the results reported will be the
average value of the data collected in all of the blocks.  If the -R
flag is used, confidence intervals will be reported to the monitor
file.
.TP 
\fB\-B\fR, \fB\-\-blocks\fR=\fIblocks[,run-time[,skip]]\fR
Use manual blocking with \fIblocks\fP blocks.  \fIblocks\fP must be
less than or equal to 30.  The run time for each block is specified
with \fIrun-time\fP.  \fISkip\fP specifies the time value of the initial
skip period.  Statistics gathered during the skip period are
discarded.  By default, its value is 0.  When the run completes, the
results reported will be the average value of the data collected in
all of the blocks.  If the \fB\-R\fP flag is used, confidence
intervals will be reported to the \fImonitor\fP file.
.TP
\fB\-C\fR, \fB\-\-confidence\fR=\fIprecision[,initial-loops[,run-time]]\fR
Use automatic blocking, stopping when the specified \fIprecision\fP is
met.  The run time of each block is estimated, based on
\fIinitial-loops\fP  running on each reference task.  The default
value for \fIinitial-loops\fP is 500.  The \fIrun-time\fP argument
specifies the maximum total run time.
.TP
\fB\-d\fR, \fB\-\-debug\fR
This option is used to dump task and entry information showing
internal index numbers.  This option is useful for determining the
names of the servers and tasks when tracing the execution of the
simulator since the Parasol output routines do no emit this
information at present.  Output is directed to stdout unless
redirected using \fB\-m \fIfile\fR.
.TP 
\fB\-e\fR, \fB\-\-error\fR=(a|i|w)\fR
Floating point error disposition.  By default, floating point errors
are ignored (\fB\-ei\fP).  Selecting \fB\-ea\fP causes \fBlqsim\fP to
abort on the floating point errors \fIoverflow, division by zero, \fP
or \fIoperand error\fP (see \fBfloating_point(3)\fP).  \fB\-ew\fP
prints a warning only on floating point error (not implemented at
present).
.IP
The solver checks for floating point overflow, division by zero and
invalid operations.  Underflow and inexact result exceptions are
always ignored.
.IP
In some instances, infinities will be propogated within the solver.
Please refer to the stop-on-message-loss pragma below.
.TP
\fB\-H\fR, \fB\-\-help\fR 
Output a summary of the command line options.
.TP
\fB\-I\fR, \fB\-\-input-format\fR=(xml|lqn)\fR
Force the input file format to be either XML or LQN.  By default, if
the input file type is one of: \fI.in\fR, \fI.lqn\fR, or \fI.xlqn\fR,
then LQN input is expected.  Otherwise XML input is expected.  
.TP
\fB\-n\fR, \fB\-\-no-execute\fR
Read input, but do not solve.  The input is checked for validity. No
output is generated. 
.TP
\fB\-o\fR, \fB\-\-output\fR=\fIfile\fR
Direct analysis results to \fIfile\fR.  A file name of `\fI\-\fR'
directs output to standard output.  If \fIoutput\fP is a directory,
all output from the simulator will be placed there with filenames
based on the name of the input files processed.  Otherwise, only one
input file can be processed; its output will be placed in
\fIoutput\fP. 
.TP
\fB\-p\fR, \fB\-\-parseable\fR
Generate parseable output suitable as input to other programs such as
\fBsrvndiff\fR and \fBmultisrvn\fR. The parseable output is sent to an
output file, one for each input file specified, determined from its
corresponding input file by the addition of \fB.p\fR.  If standard
input is used as the input source, then the parseable output is output
to the standard output device.  If the \fB-o\fI output\fR option is
used, the parseable output is sent to the file name output. (In this
case, only parseable output is emitted.)
.TP
\fB\-P\fR, \fB\-\-pragma\fR=\fIpragma[=value]\fR
Set \fIpragma\fP to value.  See the section on pragmas below.
.TP
\fB\-R\fR, \fB\-\-raw-statistics\fR
Print the values of the statistical counters to the monitor file.  If
the \fB\-A\fP, \fB\-B\fP or \fB\-C\fP option was used, the mean value, 95th and
99th percentile are reported.  At present, statistics are gathered for
the task cycle time, utilization, and waiting time for messages, and
for entry cycle time and utilization.
.TP
\fB\-S\fR, \fB\-\-seed\fR=\fInumber\fR 
Set the initial seed value for the random number generator.  By
default, the system time from time(3) is used.  The same seed value is
used to initialize the random number generator for each file when
multiple input files are specified.
.TP
\fB\-t\fR, \fB\-\-trace\fR=\fItraceopts\fR
Tracing options.
.RS
.TP
driver
Print out the underlying tracing information from the Parasol
simulation engine.
.TP
processor[=\fIpattern\fP]
Trace activity for processors whose name match \fIpattern\fP.  If
\fIpattern\fP is not specified, activity on all processors is reported.
\fIpattern\fP is regular expression of the type accepted by
\fBegrep(1)\fR.
.TP
task[=\fIpattern\fP]
Trace activity for tasks whose name match \fIpattern\fP.  If
\fIpattern\fP is not specified, activity on all tasks is reported.
\fIpattern\fP is regular expression of the type accepted by
\fBegrep(1)\fR.
.TP
events=\fIpattern[:pattern]\fP
Display only events matching pattern.  The events are: msg-async,
msg-send, msg-receive, msg-reply, msg-done, msg-abort, msg-forward,
worker-dispatch, worker-idle, task-created, task-ready, task-running,
task-computing, task-waiting, thread-start, thread-enqueue,
thread-dequeue, thread-idle, thread-create, thread-reap, thread-stop,
activity-start, activity_execute, activity_fork, and activity_join.
.TP
msgbuf
Show \fImsgbuf\fP allocation and deallocation.
.TP
timeline
Generate events for the timeline tool.
.RE
.TP
\fB\-T\fR, \fB\-\-run-time\fR=\fInumber\fR
Set the run time for the simulation.  The default is 10,000 units.
Specifying \fB\-T\fP after either \fB\-A\fP or \fB\-B\fP changes the
simulation block size, but does not turn off blocked statistics
collection.
.TP
\fB\-v\fR, \fB\-\-verbose\fR
Display solution execution.  For \fBlqsim\fP, the
simulation block number and mean confidence level are displayed.
.TP
\fB\-V\fR, \fB\-\-version\fR
Display version number.  
.TP
\fB\-w\fR, \fB\-\-no-warnings\fR
Suppress the output of warning messages.
.TP
\fB\-x\fR, \fB\-\-xml\fR
Generate XML output regardless of input format.
.TP
\fB\-\-print-interval\fR=\fInn\fP
Set the printing interval to \fInn\fP.  Results are printed after
\fInn\fP blocks have run.  The default value is 10.
.TP
\fB\-\-reload-lqx\fR
Re-run the LQX program without re-solving the models.  Results must exist from a previous solution run.
This option is useful if LQX print statements are changed.
.TP
\fB\-\-restart\fR
Re-run the LQX program without re-solving the models.  If valid results do
not exist for a given set of parameters, the simulation is run.  This option is useful if a
simulation was stopped part way through LQX execution or if some
results are invalid.
.TP
\fB\-\-debug-xml\fR
Output the XML elements and attributes as they are being parsed.  If
there is an error in the input file, parsing will generally stop at
the line causing the problem.
.TP
\fB\-\-debug-lqx\fR
Enable the debugging code for the LQX parser.  This will output the
results of the parse, not of the execution of the lqx program.
.TP
\fB\-\-print-lqx\fR
Output the LQX progam corresponding to SPEX input.
.PP
\fBLqsim\fR exits with 0 on success, 1 if the model failed to meet the
convergence criteria, 2 if the input was invalid, 4 if a command line
argument was incorrect, 8 for file read/write problems and -1 for
fatal errors. If multiple input files are being processed, the exit
code is the bit-wise OR of the above conditions.
.SH "PRAGMAS"
.TP
\fIreschedule-on-async-send = {on,off}\fP
In models with asynchronous messages, the simulator does not
reschedule the processor after an asynchronous message is sent (unlike
the case with synchronous messages).  Setting this pragma to "on"
causes a reschedule after each asynchronous message.
.TP 10
\fIscheduling\fP
.RS
.TP 15
\fIdefault\fP
Use the scheduler built into parasol for processor scheduling.  (faster)
.TP
\fIcustom\fP
Use the custom scheduler for scheduling which permits more statistics
to be gathered about processor utilization and waiting times.
However, this option invokes more internal tasks, so simulations are
slower than when using the default scheduler.
.TP
\fIdefault-natural\fP
Use the parasol scheduler; don't reschedule after the end of each
phase or activity.
.TP
\fIcustom-natural\fP
Use the custom scheduler; don't reschedule after the end of each
phase or activity.
.RE
.TP
\fImessages = nn\fP
Set the number of message buffers to \fInn\fP.  The default is 1000.
.TP
\fBspex-comment\fR=\fI{true,false}\fR
Output the model comment as a header row with SPEX output.  The default is \fIfalse\fP.
.TP
\fBspex-convergence\fR=\fIarg\fR
Set the SPEX convergence value to \fIarg\fP.  
\fIArg\fP must be a number greater than 0.
SPEX convergence only applies if SPEX the convergence section is present in the input file.
It should be set to a value with \fIless\fP precision than the convergence
used by the analytic solver and far less than the expected confidence intervals expected by
the simulator.
.TP
\fBspex-header\fR=\fI{true,false}\fR
Output the header row (column names) with SPEX output.  The default is \fItrue\fP.
.TP
\fBspex-iteration-limit\fR=\fIarg\fR
Set the SPEX Iteration Limit to \fIarg\fP.  
\fIArg\fP must be a number greater than 0.
The SPEX iteration limit only applies if SPEX the convergence section is present in the input file.
.TP
\fBspex-underrelaxation\fR=\fIarg\fR
Set the SPEX underrelaxation value to \fIarg\fP.  
\fIArg\fP must be a number between 0.0 and 1.0.
The SPEX underrelaxation only applies if SPEX the convergence section is present in the input file.
.TP
\fIstop-on-message-loss = {on,off}\fP
In models with open queueing (open arrivals or asynchronous messages),
arrival rates may exceed service rates.  The simulator can either
discard the arrival, or it can halt.
.SH "STOPPING CRITERIA"
It is important that the length of the simulation be chosen properly.
Results may be inaccurate if the simulation run is too short.
Simulations that run too long waste time and resources.
.PP
\fBLqsim\fP uses \fIbatch means\fP (or the \fImethod of samples\fP)
to generate confidence intervals.  With automatic blocking, the
confidences intervals are computed after running for three blocks plus
the initialial \fIskip\fP period.  If the root or the mean of the
squares of the confidence intervals for the entry service times is
within the specified \fIprecision\fP, the simulation stops.
Otherwise, it runs for another block and repeats the test.
With manual blocking, \fBlqsim\fP runs the number of blocks
specified then stops.  In either case, the simulator will stop after
30 blocks.
.PP
Confidence intervals can be tightened by either running additional
blocks or by increasing the block size.  A rule of thumb is the block
size should be 10,000 times larger than the largest service time
demand in the input model.
.SH "MODEL LIMITS"
The following table lists the acceptable parameter types for
\fBlqsim\fR.  An error will be reported if an unsupported parameter
is supplied unless the value is the same as the default.
.PP
.TP 32
Phases
3
.TP 
Scheduling
FIFO, HOL, PPR, RAND
.TP
Open arrivals
yes
.TP
Phase type
stochastic, deterministic
.TP
Coefficient of variation
yes
.TP
Interprocessor-delay
yes
.TP
Asynchronous connections
yes
.TP
Forwarding
yes
.TP
Multi-servers
yes
.TP
Infinite-servers
yes
.TP
Max Entries
unlimited
.TP
Max Tasks
1000
.TP
Max Processors
1000
.TP
Max Entries per Task
unlimited
.SH "SEE ALSO"
.LP
Greg Franks el. al., ``Enhanced Modeling and Solution of Layered
Queueing Networks'',
\fIIEEE Trans. Soft. Eng.\fP, Vol. 35, No. 2, Mar-Apr 2990, pp. 148-161.
.LP
\fI``The Stochastic Rendezvous Network Model for Performance of
Synchronous Multi-tasking Distributed Software''\fR by C.M. Woodside
et.al.
.LP
\fI``Layered Queueing Network Solver and Simulator User Manual''\fP.
.LP
\fI``Tutorial Introduction to Layered Modeling of Software Performance''\fP.
.LP
srvndiff(1), egrep(1), floating_point(3)
.SH BUGS
The format of the debugging output generated by the \fB\-d\fR option
requires a knowledge of the internals of \fBlqsim\fR in order to be
understood.


NEWS
* Bugs fixed in 6.2.31
468 	lqiolib 	bug-467.xlqn:48: error: Entry "WebServer" has a start activity defined. Activity "analyze" is a duplicate.
469 	lqiolib 	Reload json results from spex model fails. 
472 	lqiolib 	hexp(m,cv^2) not implemented.
466 	lqn2xxx 	terminating due to uncaught exception of type ActivityList::bad_internal_join: 
465 	lqn2xxx 	Generating .svg and .emf output is borked.
478 	lqns 		LQNS 6.2.28 handles correctly this model while 6.2.29 throws a lqns: domain error
222 	lqsim	 	V5 lqsim: faulty models causing aborts
264 	lqsim 		V6: lqsim regression.
453 	lqsim 		lqsim (V5) 63-semaphore failing. Priorities inverted in parasol.
481 	lqsim 		Processor utilization is NOT saved if the processor sharing is used.
476 	lqx 		C++20 [-Wtemplate-id-cdtor] 21 | ReferencedPointer<T>()
480 	miscella 	template-id not allowed for constructor in C++20 [-Wtemplate-id-cdtor]
470 	petrisrv 	The 'c' parameter has no effect.
423 	petrisrv 	Petrisrvn quorum borked.

* Bugs fixes in 6.2.30
208 	lqiolib 	Add --gnuplot output. 
448 	lqiolib 	Faults in open models.
456 	lqiolib 	Add drop probability for open arrivals.
458 	lqiolib 	Schema change for latest version of JMVA.
459 	lqiolib 	LQX output is not saved to a file with -ofilename 
460 	lqiolib 	Multiclass response time plotting is borked. 
384 	lqn2xxx 	Join is permitted in loop
454 	lqn2xxx 	libc++abi: terminating due to uncaught exception of type std::out_of_range: map::at: key not found
432 	lqns 		Bizarre error setting convergence value.
433 	lqns 		51-replication
452 	lqns		56-replication fails. 
449 	petrisrv 	Petrisrvn regression failures
450 	srvndiff 	srvndiff: 34-priority.lqxo:11: error: Invalid argument "-0.538461" to attribute for element <result-task>

* Bugs fixes in 6.2.29
441     lqiolib         Null model comment.
443     lqiolib         Export/Import the schema "description" field.
445     lqiolib         if ( va_list ) {}
384     lqn2xxx         Join is permitted in loop 
436     lqn2xxx         Model::format() -- bounding box is position sensitive. 
438     lqn2xxx         Queueing model output is not generated.
434     qnsolver        Bounds solution aborts if no whatif found in the input file.
446     qnsolver        Segmentation fault: in `LQX::ObjectPropertyReadNode::invoke() at SyntaxTree.cpp:2096:41

* Bugs fixed in 6.2.28
297 	lqiolib 	Gnuplot and result enhancements
343 	qnsolver 	Improve compatibility with Java Modelling Tools 
417 	qnsolver 	Bounds solution issues. 
356 	miscella 	rpm build issues.
359 	lqiolib 	Expressions are not output by Spex::print_header().
366 	lqn2xxx 	EXC_BAD_ACCESS Layer::transmorgrify() at layer.cc:571:17 [opt] 
411 	lqiolib 	JMVA issues.
412 	qnsolver 	Save JMVA WhatIf results.
413 	Document 	Missing plainurl. 	
415 	petrisrv 	Signal SIGSEGV in LQIO::DOM::Phase::getServiceTime () at dom_phase.cpp:88 
416 	lqiolib 	Fixes for plotting population mix WhatIf output. 
422 	lqiolib 	Refactor handling of control variables. 
366 	lqn2xxx 	EXC_BAD_ACCESS Layer::transmorgrify() at layer.cc:571:17 [opt] 
428 	lqn2xxx 	lqn2fig: EXC_BAD_ACCESS libsystem_platform.dylib`_platform_strlen + 4
432 	lqns	 	Bizarre error setting convergence value. 	
433 	lqns	 	51-replication
425 	lqns	 	Refactor initPopulation. 
430 	lqns 		activity-templ.lqn:75: error: Task "Server", and-join "parB & parA" does not match and-fork "parA & parB" at line 74.
418 	lqsim 		lqsim crashes running -t... 	
419 	qnsolver 	model.cc:(.text+0x18ff): undefined reference to `typeinfo for QNIO::JMVA_Document'

* Bugs fixed in 6.2.27
192 	libmva 	 	MVA::marginalProbabilities2 mis-dimensioned. 
356 	miscella 	rpm build issues.
373 	lqns 	 	Processor waiting times at Rolia multiserver make no sense. 
380 	lqns 	 	lqns hangs with Conway multiserver. 
382 	lqn2xxx 	lqn2ps -Oout doesn't print processors. 
383 	lqiolib 	<pre> precedences can accept more than one activity in XML (JSON?) input. 
385 	petrisrv 	V5 petrisrvn: faulty models causing aborts.
386 	lqn2csv 	getopt_long headaches. 
387 	srvndiff 	File glob not working. 
389 	srvndiff 	C:\msys64\mingw32\bin\srvndiff.exe: model-001.lqxo:12: error: unclosed token 
390 	lqn2xxx 	lqn2xxx: command not found. 
391 	lqsim 		lqsim/regression/Makefile.am, 5*-replication. 
392 	petrisrv 	Calls are not being constructed
393 	lqiolib 	Save (and print) marginal queue probabilities.
395 	petrisrv 	V5 petrisrvn: faulty models causing aborts or failing.
397 	lqiolib 	Add qnap2 parser for BCMP models.
398 	lqiolib 	Refactor LQX math handing from ExternalVariables to LQX Expressions.
399 	lqiolib 	WhatIf construction is bogus.
400 	lqn2xxx 	Converting from json to lqn fails on Windows.
401 	libmva 		libc++abi: terminating with uncaught exception of type std::invalid_argument: default-output
402 	miscella 	Mac M1 issues.
403 	libmva 	 	43-activities.lqnx: Floating point exception: mva.cc 541: Inexact result, Invalid operation
405 	lqns 	 	Optimizer bug: option.h
406 	libmva 	 	Refactor fpgoop.
407 	lqns 	 	--srvn-layering crashes in V6 interlocking code
409 	petrisrv 	Petrisrvn fails when SPEX/LQX is used to alter the arrival rate to an entry. 
411 	lqiolib 	JMVA issues. 

* Bugs fixed in 5.26
365 	lqns	 	The warning for zero service time should be suppressed for reference tasks with think time. 
369 	petrisrv 	Segmentation violation with --overtaking 
370 	lqiolib 	Activity parameters probabilities and rates are constants. 
379 	lqns	 	V5 Regression. 	
372 	lqiolib 	Result variables can be reused. 
374 	lqns 		Asynchronous Cycles #1 	
375 	lqns 		Asynchronous Cycles #1 	
376 	lqn2xxx 	Overloaded infinite servers with open arrivals... 
378 	lqn2xxx 	Activity throughput not initialized. 

* Bugs fixed in 5.25
338 	lqns 		Queuing time not equal to zero when NCust = NServers for Zhou multiserver. 
352 	lqn2xxx 	--no-xxxx doesn't work. 
353 	lqns 		Assertion failed: (ix && ix <= sz), function operator[], file vector.h, line 174. 

doc/userman/auto/lqsim.el
(TeX-add-style-hook "lqsim"
 (lambda ()
    (LaTeX-add-index-entries
     "Parasol"
     "output!parseable"
     "input!XML"
     "XML"
     "output!XML"
     "command line"
     "automatic blocking"
     "block!automatic"
     "block!size"
     "simulation!block"
     "block!simulation"
     "run time!simulation"
     "precision!simulation"
     "confidence level"
     "skip period"
     "statistics"
     "simulation!statistics"
     "confidence intervals"
     "file!monitor"
     "monitor file"
     "manual blocking"
     "block!manual"
     "skip"
     "initial-loops"
     "task!reference"
     "reference task"
     "floating point!exception"
     "overflow"
     "infinity"
     "floating point!infinity"
     "service time!distribution"
     "distribution!service time"
     "output!csv"
     "file!debug"
     "file!tracing"
     "output"
     "standard input"
     "pragma"
     "counters!statistical"
     "statistical counters"
     "cycle-time!task"
     "cycle-time!entry"
     "utilization!task"
     "utilization!processor"
     "utilization!entry"
     "waiting time"
     "seed"
     "random number!generation"
     "tracing"
     "processor!trace"
     "trace!processor"
     "task!trace"
     "trace!task"
     "statistics!simulation"
     "solution!statistics"
     "version"
     "copyright"
     "warning!ignore"
     "delay!interprocessor"
     "LQX"
     "LQX!debug"
     "XML!debug"
     "exit!success"
     "convergence!failure"
     "input!invalid"
     "command line!incorrect"
     "error!fatal"
     "input!multiple"
     "environment variable"
     "LQSIM\\_PRAGMAS@\\texttt{LQSIM\\_PRAGMAS}"
     "environment
  variable!override"
     "pragma!invalid!command line"
     "pragma!invalid!input file"
     "processor!scheduling"
     "processor!scheduling!custom"
     "processor!utilization"
     "waiting!processor"
     "processor!waiting"
     "processor!scheduling!natural"
     "reschedule!phase"
     "phase!reschedule"
     "reschedule!activity"
     "activity!reschedule"
     "message!buffers"
     "arrival rate"
     "open arrival!overflow"
     "send-no-reply!overflow"
     "send-no-reply"
     "stopping criteria"
     "length!simulation"
     "batch means"
     "method of samples"
     "root mean square"
     "service time!entry"
     "service time!demand"
     "phase!maximum"
     "scheduling"
     "open arrival"
     "phase!type"
     "think time"
     "coefficient of variation"
     "interprocessor delay"
     "asynchronous connections"
     "forwarding"
     "multi-server"
     "infinite server"
     "entry!maximum"
     "task!maximum"
     "processor!maximum"
     "limits!lqsim")
    (LaTeX-add-labels
     "sec:invoking-lqsim"
     "sec:lqsim-options"
     "sec:lqsim-return-status"
     "sec:lqsim-pragmas"
     "sec:stopping-criteria"
     "sec:lqsim-model-limits"
     "tab:lqsim-model-limits")))

